examples/mdetr/model.py

```diff
@@ -1,370 +0,0 @@
-        if include_contrastive:
-            if not self._is_eligible_for_contrastive(tokenized):
-                raise ValueError(
-                    "Module is not eligible to calculate contrastive alignment loss"
-                )
-
-            contrastive_query_embeddings = F.normalize(
-                self.contrastive_alignment_image_projection(final_hidden_state),
-                p=2,
-                dim=-1,
-            )
-            contrastive_token_embeddings = F.normalize(
-                self.contrastive_alignment_text_projection(
-                    model_output.transformer_output.text_memory
-                ).transpose(0, 1),
-                p=2,
-                dim=-1,
-            )
-        else:
-            contrastive_query_embeddings, contrastive_token_embeddings = None, None
-
-        # Apply VQA heads to get answer predictions
-        answer_preds = self.vqa_heads(model_output.extra_embeddings.transpose(0, 1))
-
-        target_boxes = [t["boxes"] for t in targets]
-        # Get the matching between predicted and target boxes
-        indices = self.matcher(
-            model_output.pred_logits,
-            model_output.pred_boxes,
-            target_boxes,
-            positive_map,
-        )
-
-        # Calculate MDETR loss with VQA losses
-        loss = self.loss(
-            model_output.pred_logits,
-            model_output.pred_boxes,
-            targets,
-            positive_map,
-            indices,
-            contrastive_query_embeddings,
-            contrastive_token_embeddings,
-            tokenized,
-            answer_preds,
-            answers,
-            answer_types,
-            weight_dict,
-        )
-
-        return MDETRVQAOutput(model_output, answer_preds, loss)
-
-
-def mdetr_for_vqa(
-    num_queries: int = 100,
-    num_classes: int = 255,
-    embedding_dim: int = 768,
-    transformer_d_model: int = 256,
-    transformer_num_heads: int = 8,
-    transformer_encoder_layers: int = 6,
-    transformer_decoder_layers: int = 6,
-    transformer_dim_feedforward: int = 2048,
-    transformer_dropout: float = 0.1,
-    return_intermediate_dec: bool = True,
-    matcher_cost_class: int = 1,
-    matcher_cost_bbox: int = 5,
-    matcher_cost_giou: int = 2,
-    no_object_weight: float = 0.1,
-    contrastive_dim: Optional[int] = None,
-    temperature: Optional[float] = None,
-):
-    hidden_dim = transformer_d_model
-    vqa_heads = mdetr_gqa_heads()
-    num_heads = len(vqa_heads.heads.keys())
-
-    model = mdetr_resnet101(
-        num_queries,
-        num_classes,
-        embedding_dim,
-        transformer_d_model,
-        transformer_num_heads,
-        transformer_encoder_layers,
-        transformer_decoder_layers,
-        transformer_dim_feedforward,
-        transformer_dropout,
-        return_intermediate_dec,
-        num_extra_query_embeddings=num_heads,
-    )
-    if contrastive_dim is not None:
```

Comment:
@ankitade in response to your question about strict checkpoint loading: the previous version of MDETRForVQA had optional contrastive projections, while the new one makes them mandatory. But these projections are not present in the GQA fine-tuned checkpoint. So we need to disable strict checkpoint loading to do VQA eval. The alternative is to make them optional again, but it makes the code kinda messy